docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+
+Sub-categories
+-------------------------
+
+Smaller mods will generally have fewer properties (say about 5 - 10) and do not need to worry about dividing up their configurations into more manageable groups of properties for the end-user to manage. However, for larger mods configurations consisting of 20 to 50 elements is not all that uncommon, and the need for sub-categories becomes irrefutable. Recognizing such a need, the annotation-based configuration system provides a mechanism by which `@Config` classes can be divided into groups that are reflected directly in the structure of the `@Config` class.
+
+The sub-division of `@Config` classes is accomplished by the addition of `public static` inner classes that act as the physical configuration groupings. Thus, any property that is placed in the inner class is then resolved by the system as not to exist in the default group defined by the `@Config` annotation but rather a new group with the name of the inner class.
+
+Building upon the prior examples, an example usage of property grouping is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property belongs to the default group.")
+  public static boolean myPropertyDefaultGroup = false;
+
+  public static class MyCategory {
+
+    @Comment("This property belongs to the MyCategory group.")
+    public static boolean myPropertyMyCategory = false;
+
+  }
+
+}
+````
+
+
+Alternate Names
+-------------------------
+
+Sometimes, for reasons unknown, we wind up with variable names in a configuration class that would be unsuitable for use as an actual property key. This would be problematic as Forge determines the property key based off of variable names. For this reason, the `@Name` annotation exists.
+
+The name annotation allows for the convenient changing of the property's name from the default value inferred from the name of the variable to something that is deemed more suitable by the modder.
+
+An example usage would be as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property would normally have a horrible name!")
+  @Name("my_well_named_property")
+  public static boolean MY_hoRibbliyNAMEdVairable = false;
+
+}
+````
+
+!! Note
+
+    If you are following conventional Java variable naming practices then the variables defined in your `@Config` class likely contain a great number of capital letters, as seen in the provided example usages (prior to the horribly named one). Since the Forge standard for registration names is all-lowercase, the default property names would therefore be unsuitable, and use of the `@Name` annotation is highly recommended.
```

Comment:
"conventional" is subjective. My conventions are likely different from your conventions, which are likely different from tterrag's conventions, which are likely different from the conventions specified by Oracle.

Reply:
Java convention is well-defined as per [The Java Language Specification](https://docs.oracle.com/javase/specs/jls/se8/html/index.html). See Chapter 6 for an in-depth description of the Java naming system and associated convention.

Reply:
The Java "official" convention is still subjective and many people don't use it because they don't like it or haven't read it. Personally I've never even read through it in its entirety and I've been using Java for about half a decade now (I based my code style off of Google's standards roughly, if I recall correctly).